source/opt/inline_pass.h

```diff
@@ -139,9 +176,21 @@ class InlinePass : public Pass {
   // Map from block's label id to block.
   std::unordered_map<uint32_t, ir::BasicBlock*> id2block_;
 
-  // Set of ids of inlinable function 
+  // Set of ids of functions with early returns
+  std::set<uint32_t> early_return_;
+
+  // Set of ids of functions with no returns in loop
+  std::set<uint32_t> no_return_in_loop_;
+
+  // Set of ids of inlinable functions
   std::set<uint32_t> inlinable_;
 
+  // Map from block to its structured successor blocks
```

Comment:
Yes, that's fine.  Debatable where the definition should go.    They're close enough to each other that it doesn't matter to me.